src/gt4py/eve/extended_typing.py

```diff
@@ -526,6 +526,14 @@ def is_Any(obj: Any) -> bool:
 
 def has_type_parameters(cls: Type) -> bool:
     """Return ``True`` if obj is a generic class with type parameters."""
+    if len(getattr(cls, "__parameters__", [])) == 0:
+        cls.__parameters__ = []
+        annotations = getattr(cls, "__annotations__", {})
+        type_ = annotations.get("type")
+        if annotations and type_:
+            for t in get_args(type_):
+                if isinstance(t, _typing.TypeVar):
+                    cls.__parameters__.append(t)
```

Comment:
I don't really understand these lines, but they are adding a new `__parameters__` member to any class passed as parameter, which is effectively monkey-patching the class for no reason.

Reply:
Hi @egparedes ,
FYI, I checked the python 3.10 and the `__parameters__` member is missing as well. So it seems to be a recurring issue not related to 3.11 only. It does not crash because it never enters this part of the code. I tried to understand where they diverge (3.10 vs 3.11) but I could not find it.
Essentially what the above code does is to check if this field exists and if not to properly construct it. Another place that would make sense to put it, would be this part of the code: https://github.com/kotsaloscv/gt4py/blob/py311_fix/src/gt4py/eve/datamodels/core.py#L1071
What do you think?